raft_test.go

```diff
@@ -212,11 +211,16 @@ func TestRaft_RecoverCluster(t *testing.T) {
 		c.EnsureSame(t)
 		c.EnsureSamePeers(t)
 	}
-	for applies := 0; applies < 10; applies++ {
-		t.Run(fmt.Sprintf("%d applies", applies), func(t *testing.T) {
-			runRecover(t, applies)
-		})
-	}
+
+	t.Run("no snapshot, no trailing logs", func(t *testing.T) {
+		runRecover(t, 0)
+	})
+	t.Run("snapshot", func(t *testing.T) {
+		runRecover(t, snapshotThreshold)
+	})
+	t.Run("snapshot, with trailing logs", func(t *testing.T) {
+		runRecover(t, snapshotThreshold+20)
+	})
```

Comment:
The old comment said:

>  By sweeping through the trailing logs value we will also hit the case where we have a snapshot only.

But I'm not sure what that means. We never set the snapshot threshold previously. Is it the `TrailingLogs` value that is relevant here? I don't see how. It's only used in the code to prune.

Reply:
I've made one change to this, so that the second case is "some logs with no snapshot", which I think is a more correct description of that case. Also changed the applies value to `snapshotThreshold-1` to make sure we don't have a snapshot yet.